src/hats/io/file_io/file_io.py

```diff
@@ -310,14 +308,22 @@ def read_parquet_file_to_pandas(file_pointer: str | Path | UPath, **kwargs) -> n
     file_pointer = get_upath(file_pointer)
     # If we are trying to read a directory over http, we need to send the explicit list of files instead.
     # We don't want to get the list unnecessarily because it can be expensive.
-    if isinstance(file_pointer, upath.implementations.http.HTTPPath) and file_pointer.is_dir():
+    if (
+        isinstance(file_pointer, upath.implementations.http.HTTPPath) and file_pointer.is_dir()
+    ):  # pragma: no cover
         file_pointers = [f for f in file_pointer.iterdir() if f.is_file()]
         return npd.read_parquet(
             file_pointers,
             filesystem=file_pointer.fs,
             partitioning=None,  # Avoid the ArrowTypeError described in #367
             **kwargs,
         )
+    use_bytes_io = _parquet_bytes_io(file_pointer)
+    if use_bytes_io:  # pragma: no cover
+        req_info = urllib.request.Request(file_pointer.path)
+        with urllib.request.urlopen(req_info) as req:
+            reader = BytesIO(req.read())
+            return npd.read_parquet(reader, partitioning=None, **kwargs)
```

Comment:
This repeated fragment is just a little larger than an idiom, and appears three times.  It could be clearer to have it be a wrapper, e.g. `return _cache_all_bytes(npd.read_parquet, **kwargs)` and similarly throughout.